js/web/lib/wasm/jsep/backend-webgpu.ts

```diff
@@ -234,19 +263,92 @@ export class WebGpuBackend {
   }
 
   flush(): void {
-    if (this.commandEncoder) {
-      this.endComputePass();
-      this.device.queue.submit([this.getCommandEncoder().finish()]);
-      this.gpuDataManager.refreshPendingBuffers();
-      this.commandEncoder = null;
-      this.pendingDispatchNumber = 0;
+    if (!this.commandEncoder) {
+      return;
     }
-  }
 
-  isQueryEnabled(): boolean {
-    return this.device.features.has('timestamp-query') &&
-        (this.env.webgpu.profiling?.mode === 'default' ||
-         (!this.env.webgpu.profiling?.mode && this.env.webgpu.profilingMode === 'default'));
+    TRACE_FUNC_BEGIN();
+    let queryReadData: GpuData;
+    if (this.queryType !== QueryType.none) {
+      this.commandEncoder.resolveQuerySet(
+          this.querySet!, 0, this.pendingDispatchNumber * 2, this.queryResolveData!.buffer, 0);
+      queryReadData = this.gpuDataManager.create(
```

Comment:
If we want to keep the GPU buffer consistency between using query vs not using query, we may want to create/release buffer without using GpuDataManager.

When using `gpuDataManager.create()` and `gpuDataManager.release()` to manage the GPU buffer for query usage, it may change how GPU backend deal with tensor buffer caches.

Reply:
Done and this is really a good point! I'm refactoring the buffer management part, and I will unify the usages so that it's easy to output the total memory consumption at multiple stages. I will keep this rule in mind.